### DISCUSSION TITLE
How to contribute code to the CLI package?

### DISCUSSION THREAD

**Discussion by willpinha** [huntabyte/shadcn-svelte#529] [Help] [answered]
Hey! I have already read the CONTRIBUTING guide. I want to resolve a CLI package issue, so I forked the repository to test locally. However, I am finding it difficult to understand how to test the CLI. Is there any guide or explanation on how to do this (which commands to run, the workflow to test the CLI, ...)?

**Comment by willpinha** [comment]
I thought about using `npm link` to test the CLI locally through another project, but I don't know if that's the way to test it

**Comment by babakfp** [comment]
Hey

In the forked/cloned workspace, navigate to `packages/cli`. Take a look at the `README.md` and `package.json` files in that subdirectory. Use the `dev` command for development (compiling ts). Use `start:dev` or `start` commands to run the cli.

**Reply by willpinha** [reply]
Makes sense, thanks! My only question is that running `start:dev` in the `packages/cli` directory causes the CLI to look for files in the directory itself (like `tailwind.config.js`, or `.svelte-kit/tsconfig.json` ).

So what is the way to test the CLI in other projects locally? I used `npm link` to test my changes in another local project (like a SvelteKit project), to see if the CLI is working as expected.

Additionally, I'm unsure about how the code will be tested by whoever is reviewing my pull request, as no automatic tests are being run for the CLI.

**Reply by babakfp** [reply, marked as answer]
I don't use the `npm link` feature myself. Instead, I create a new project (a playground), in the root directory, where the other project is located, like this:

```
| my-root-folder
| ---- shadcn-svelte/packages/cli
| ---- my-test-playground
```

I run the `start:dev` command like this `cross-env COMPONENTS_REGISTRY_URL=http://localhost:5173 node ../shadcn-svelte/packages/cli/dist/index.js`.

I don't know about tests.
